src/plugins/memdump/memdump.cpp

```diff
@@ -291,6 +293,122 @@ bool dump_memory_region(
     return ret;
 }
 
+static event_response_t terminate_process_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    // HANDLE ProcessHandle
+    uint64_t process_handle = drakvuf_get_function_argument(drakvuf, info, 1);
+
+    if (process_handle != 0xffffffffffffffffULL)
+    {
+        PRINT_DEBUG("[MEMDUMP] Process handle not pointing to self, ignore\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    auto plugin = get_trap_plugin<memdump>(info);
+    if (!plugin)
+        return VMI_EVENT_RESPONSE_NONE;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    access_context_t ctx =
+    {
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3
+    };
+
+    bool is_32bit;
+    addr_t stack_ptr;
+    addr_t frame_ptr;
+
+    if (drakvuf_get_user_stack32(drakvuf, info, &stack_ptr, &frame_ptr))
+    {
+        is_32bit = true;
+    }
+    else if (drakvuf_get_user_stack64(drakvuf, info, &stack_ptr))
+    {
+        is_32bit = false;
+    }
+    else
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to get stack pointer\n");
+        drakvuf_release_vmi(drakvuf);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    PRINT_DEBUG("[MEMDUMP] Got stack pointer: %llx\n", (unsigned long long)stack_ptr);
+
+    size_t bytes_read = 0;
+    uint8_t buf[512];
+    ctx.addr = stack_ptr;
+    vmi_read(vmi, &ctx, 512, buf, &bytes_read);
```

Comment:
Check return status

Reply:
Doesn't VMI return `VMI_FAILURE` when it is not possible to read full 512 bytes? That's why the following `for` loop reads up to `bytes_read` bytes. It is not required for this to read the whole buffer.

Reply:
OK, in that case add a comment explaining that it's on purpose